### DISCUSSION TITLE
WebRTC firewall rules

### DISCUSSION THREAD

**Discussion by lurendrejer** [Ylianst/MeshCentral#6524] [General]
Hi.

We have a quite restrictive network setup, I have set up WebRTC in mesh - which works when used from a network allowing any traffic, to any networks.

What specific ports / protocols should be allowed to allow WebRTC to and from the more restrictive networks?

**Comment by si458** [comment]
webrtc doesnt use set port numbers,
it will use whatever port it feels like, and it will even use UDP if it can too

**Reply by lurendrejer** [reply]
Yup, an ability to limit the 'random high port' functionality of webrtc would preferable in a tool like meshcentral.

**Reply by si458** [reply]
but i dont believe the is a way because webrtc is built into the browser and its the browser that sets/picks the port
unless u can provide docs which explain HOW to tell a browser in javascript to use a port between X and Y
its not going to be possible sadly

**Reply by lurendrejer** [reply]
ah... that makes sense.

**Reply by lurendrejer** [reply]
As far as i can see, it is not possible in the browser - but the 'mediaserver' in RTC terminology should be able to limit the portrange.
So, i the mesh-agent should be in control.

**Reply by si458** [reply]
then this would require extensive reconfig/recompile of the agent to support this 
and wont be easy!

**Reply by lurendrejer** [reply]
The 'turn' server can also dictate this. I don't know if the meshcentral-server is an actual TURN server or something propriatary.

**Reply by si458** [reply]
meshcentral isnt a TURN server, its just the main server to setup the webrtc,
the client and agent talk directly to each other over webrtc IF POSSIBLE, and if not possible then just use the standard websockets which get relayed via your meshcentral server
